### PR TITLE
nodes 2.0 controller and unit tests

### PR DESCRIPTION
**Pull request by abildw**
@RackHD/corecommitters @dalebremner @geoff-reid @srinia6

### CODE REVIEW

**Comment by jlongever** [comment]
test this please

**Comment by jlongever** [comment]
@abildw Could you check this error out?

*** BUILD #372 ***
Test Name: test_node_id_obm
Error Details: (500)
Reason: Internal Server Error
HTTP response headers: HTTPHeaderDict({'Content-Length': '46', 'X-Powered-By': 'Express', 'Connection': 'keep-alive', 'ETag': 'W/"2e-zmsfLEJtfszmxVweCidQvA"', 'Date': 'Fri, 12 Feb 2016 23:13:50 GMT', 'Access-Control-Allow-Origin': '*', 'Content-Type': 'application/json; charset=utf-8'})
__HTTP response body: {"message":"Maximum call stack size exceeded"}__

**Comment by brianparry** [comment]
@abildw This looks like the problem that @geoff-reid was hitting earlier.  Did you get his fix?

File: lib/api/1.1/northbound/nodes.js [hunk 1]
```diff
@@ -307,34 +307,6 @@ function nodesRouterFactory (
...
-    /**
-     * @api {post} /api/1.1/nodes/:macaddress/dhcp/whitelist POST /:mac/dhcp/whitelist
-     * @apiVersion 1.1.0
-     * @apiDescription Post whitelist.
-     * @apiName node-dhcp-whitelist-post
-     * @apiGroup nodes
-     * @apiParam {String} macaddress macaddress of node
-     */
-
-    router.post('/nodes/:macaddress/dhcp/whitelist', rest(function(req) {
```
Comment (jlongever) [inline lib/api/1.1/northbound/nodes.js]: @abildw Was removing this intentional?

File: lib/api/1.1/northbound/nodes.js [hunk 2]
```diff
@@ -307,34 +307,6 @@ function nodesRouterFactory (
...
-
-     * @api {delete} /api/1.1/nodes/:macaddress/dhcp/whitelist DELETE /:mac/dhcp/whitelist
-     * @apiVersion 1.1.0
-     * @apiDescription Remove a whitelist of specified mac address
-     * @apiName node-dhcp-whitelist-delete
-     * @apiGroup nodes
-     * @apiParam {String} macaddress macaddress of node
-     */
-
-    router.delete('/nodes/:macaddress/dhcp/whitelist', rest(function(req) {
```
Comment (jlongever) [inline lib/api/1.1/northbound/nodes.js]: @abildw Was removing this intentional?

**Comment by jlongever** [comment]
-1 Looks like this breaks 1.1 backwards compatibility.

**Comment by dalebremner** [comment]
:+1:

**Comment by brianparry** [comment]
:+1:

**Comment by jlongever** [comment]
+1

**Pull request merged**